apps/ios/Shared/Views/Chat/ChatItemForwardingView.swift

```diff
@@ -0,0 +1,145 @@
+                    }
+                }
+        }
+    }
+
+    @ViewBuilder private func forwardListView() -> some View {
+        VStack(alignment: .leading) {
+            let chatsToForwardTo = filterChatsToForwardTo()
+            if !chatsToForwardTo.isEmpty {
+                ScrollView {
+                    LazyVStack(alignment: .leading, spacing: 8) {
+                        searchFieldView(text: $searchText, focussed: $searchFocused)
+                            .padding(.leading, 2)
+                        let s = searchText.trimmingCharacters(in: .whitespaces).localizedLowercase
+                        let chats = s == "" ? chatsToForwardTo : chatsToForwardTo.filter { $0.chatInfo.chatViewName.localizedLowercase.contains(s) }
+                        ForEach(chats) { chat in
+                            Divider()
+                            forwardListNavLinkView(chat)
+                                .disabled(chatModel.deletedChats.contains(chat.chatInfo.id))
+                        }
+                    }
+                    .padding(.horizontal)
+                    .padding(.vertical, 8)
+                    .background(Color(uiColor: .systemBackground))
+                    .cornerRadius(12)
+                    .padding(.horizontal)
+                }
+                .background(Color(.systemGroupedBackground))
+            } else {
+                emptyList()
+            }
+        }
+    }
+
+    private func filterChatsToForwardTo() -> [Chat] {
+        var filteredChats = chatModel.chats.filter({ canForwardToChat($0) })
+        if let index = filteredChats.firstIndex(where: { $0.chatInfo.chatType == .local }) {
+            let privateNotes = filteredChats.remove(at: index)
+            filteredChats.insert(privateNotes, at: 0)
+        }
+        return filteredChats
+    }
+
+    private func canForwardToChat(_ chat: Chat) -> Bool {
+        switch chat.chatInfo {
+        case let .direct(contact): contact.sendMsgEnabled && !contact.nextSendGrpInv
+        case let .group(groupInfo): groupInfo.sendMsgEnabled
+        case let .local(noteFolder): noteFolder.sendMsgEnabled
+        case .contactRequest: false
+        case .contactConnection: false
+        case .invalidJSON: false
+        }
+    }
+
+    private func emptyList() -> some View {
+        Text("No filtered chats")
+            .foregroundColor(.secondary)
+            .frame(maxWidth: .infinity)
+    }
+
+    @ViewBuilder private func forwardListNavLinkView(_ chat: Chat) -> some View {
+        Button {
+            if chat.id != fromChatInfo.id {
+                Task {
+                    await MainActor.run {
+                        chatModel.forwardToChatId = chat.id
+                        chatModel.forward = ComposeState.init(forwardingItem: ci, fromChatInfo: fromChatInfo)
+                        dismiss()
+                    }
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
the problem can be that it is executed after Task termination - why it is asyncAfter and not Task.delay with await MainActor.run?